packages/server/src/fhir/repo.ts

```diff
@@ -1000,8 +1002,8 @@ export class Repository extends BaseRepository implements FhirRepository<PoolCli
    * @param builder - The select query builder.
    */
   private addProjectFilters(builder: SelectQuery): void {
-    if (this.context.project) {
-      builder.where('compartments', 'ARRAY_CONTAINS', [this.context.project, r4ProjectId], 'UUID[]');
+    if (this.context.projects?.length) {
+      builder.where('compartments', 'ARRAY_CONTAINS', [...this.context.projects, r4ProjectId], 'UUID[]');
```

Comment:
Ok, i wondered about this -- Here we're going to just always add `r4ProjectId` at filter time.

Would it make more sense to add `r4ProjectId` to `this.context.projects` in the constructor?